libs/ledger/src/storage_unit/storage_unit_client.cpp

```diff
@@ -144,43 +144,31 @@ bool StorageUnitClient::RevertToHash(Hash const &hash, uint64_t index)
 
   if (genesis_state && (index == 0))  // this is truly the genesis block
   {
-    FETCH_LOG_INFO(LOGGING_NAME, "Reverting state to genesis.");
+    FETCH_LOG_INFO(LOGGING_NAME, "Reverting state to genesis!");
+  }
 
-    // fill the tree with empty leaf nodes
-    for (std::size_t i = 0; i < num_lanes(); ++i)
-    {
-      tree[i] = chain::GENESIS_MERKLE_ROOT;
-    }
+  uint64_t const merkle_stack_size = permanent_state_merkle_stack_.size();
```

Comment:
The old logic here was wrong. If you were to revert to genesis it incorrectly assumed that the lanes would all be GENESIS_MERKLE_ROOT - not true.